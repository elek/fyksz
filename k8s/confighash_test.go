package k8s

import (
	"io"
	"testing"
)

func TestConfigHash(t *testing.T) {
	WithInputOutput(t, "confighash_test.input.yaml", "confighash_test.expected", func(input io.Reader, output io.Writer) error {
		r := ConfigHash{
			input:  input,
			output: output,
		}
		return r.Run()
	})
}
