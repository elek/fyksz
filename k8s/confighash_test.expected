apiVersion: apps/v1
kind: Deployment
metadata:
  name: storagenode01
  annotations:
    fyksz-config-hash/hdfs-config: 6844298a2c07e301a0ce6e31d26e5bac
spec:
  replicas: 1
  strategy:
    type: Recreate
  selector:
    matchLabels:
      app: storagenode01
  template:
    metadata:
      labels:
        app: storagenode01
    spec:
      containers:
      - image: foobar
      volumes:
      - name: config
        configMap:
          name: hdfs-config

---
apiVersion: v1
kind: ConfigMap
metadata:
  name: hdfs-config
data:
  foo: bar
